rust/blockstore/src/arrow/block/types.rs

```diff
@@ -66,88 +66,119 @@ impl Block {
         delta
     }
 
-    fn binary_search<'me, K: ArrowReadableKey<'me>, V: ArrowReadableValue<'me>>(
+    /// Binary search the blockfile to find the partition point of the specified prefix and key.
+    /// The implementation is based on [`std::slice::partition_point`].
+    ///
+    /// `(prefix, key)` serves as the search key, and it is sorted in ascending order.
+    /// The partition is defined by: `|x| x < (prefix, key)`.
```

Comment:
nit: It's not clear from the description that it is searching for equal prefix. IMO something like "the partition point is the first x s.t. x.prefix == target.prefix and x.key >= target.key" is more clear

Reply:
In rust std the partition predicate should return true on the first half of the slice and false on the second half of the slice. The partition point is the start of the second half. I'll add a comment to clarify this